qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java

```diff
@@ -87,7 +87,7 @@ static List<HttpHost> parseHosts(String props) {
     }
 
     public static void configureRemoteClusters(List<Node> remoteNodes) throws Exception {
-        assertThat(remoteNodes, hasSize(3));
```

Comment:
I don't think `assertBusy` will help you here? The list is just created one by a call to `getNodes(..` so it won't ever change?

Reply:
You are right. It seems environment-dependent, and applying something like the following minimizes the possibility of the assert failing by adding extra cycles to the execution. Well, I cannot think of an alternative if there are network calls in between.

```
 final List<Node>[] remoteNodes = new List<Node>[1];
        assertBusy(() -> {
            remoteNodes[0] = getNodes(restClient);
            assertThat(remoteNodes[0], hasSize(3));
        });
```